### DISCUSSION TITLE
Can I use ISender or IMediator on Razor page (PageModel) ?

### DISCUSSION THREAD

**Discussion by siwaphatpg** [jasontaylordev/CleanArchitecture#1150] [General]
I'm new of using Clean Architecture. 
But I think I have understood all the main concepts of this super useful architecture.

My problem is I'm using version 8 and I selected WebAPI only project because I wanted to use just "cshtml" for front end development.
There are any examples of Clean Architecure with my requirement ?

**Comment by neozhu** [comment]
Yes, Razor Pages and MVC are very similar in their functionality and usage.

**Reply by jasontaylordev** [reply]
Are you looking to do Razor Pages or MVC?

**Reply by siwaphatpg** [reply]
Thanks for your responce.
I have just figured out how to use MVC with ISender.

![ภาพ](https://github.com/jasontaylordev/CleanArchitecture/assets/18515311/df0b80a8-2f74-4905-8044-0b7945bcfcdc)

I created a base class for my controller and It has a ISender property.
Now I can send a query / command like I what I do on API controller.

Please tell me if I did something wrong.

**Reply by neozhu** [reply]
A long time ago, I used this framework in a Razor Pages project.  refer to https://github.com/neozhu/RazorPageCleanArchitecture
for example,  
```c#
 [Authorize(policy: Permissions.Documents.View)]
    public class IndexModel : PageModel
    {
        [BindProperty]
        public AddEditDocumentCommand Input { get; set; }
        [BindProperty]
        public IFormFile UploadedFile { get; set; }
        public SelectList DocumentTypes { get; set; }

        private readonly ISender _mediator;
        private readonly IStringLocalizer<IndexModel> _localizer;

        public IndexModel(
                ISender mediator,
            IStringLocalizer<IndexModel> localizer
            )
        {
            _mediator = mediator;
            _localizer = localizer;
        }
        public async Task OnGetAsync()
        {

            var request = new DocumentTypesGetAllQuery();
            var documentTypeDtos = await _mediator.Send(request);
            DocumentTypes = new SelectList(documentTypeDtos, "Id", "Name");
        }
}
```

**Reply by siwaphatpg** [reply]
I'm familiar with MVC than razor page.
Thanks for your example source code. I'm familiar with MVC than razor page.
Your example make me easy to get started on developing Razor page application.
I have given a star.
